projects/js-packages/publicize-components/src/social-store/actions/connection-data.js

```diff
@@ -83,6 +83,22 @@ export function mergeConnections( freshConnections ) {
 			toggleable: true,
 		};
 
+		// A connection was added after the test call started, so we need to add it to the list.
+		if ( prevConnections.length > freshConnections.length ) {
+			const addedConnection = prevConnections.find(
+				conn =>
+					! freshConnections.some( freshConn =>
+						freshConn.connection_id
+							? freshConn.connection_id === conn.connection_id
+							: freshConn.id === conn.id
+					)
+			);
+
+			if ( addedConnection ) {
+				connections.push( addedConnection );
+			}
+		}
+
```

Comment:
This approach is problematic. Currently, if a post has connections saved in post meta (`jetpack_publicize_connections`), it's possible that the list is outdated and any of those connections would have been removed from connections management. Thus, connection test results resets the list and ensures that only the active connections are in the list.

With this change, it will add those removed connections back to the list which is not what we want.

Steps:
- Add some connections
- Create a post and publish or save it as draft.
- Goto Social or Jetpack settings page and remove one or more connections
- Go back to the post editor
- The removed connections should not be retained after test results.